.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname $0)/_/husky.sh"
```

Comment:
why do we need this? the docs don't mention it

Reply:
This is added automatically when running `npx husky add ...` to add a new hook.

Installing husky creates a "hidden" directory `_` that contains `husky.sh` that does some error handling and friendly error messages. Husky now gets triggered by git config, instead of previously adding shell scripts in the `.git/hooks`.

In the end, those lines are automatically added by husky. The actual `husky.sh` file is not intended to be checked in. They seem to have completely reworked their integration in version 5.

Reply:
understood, thanks. I was asking as this is not shown in the docs